azkaban-hadoop-security-plugin/build.gradle

```diff
@@ -1,12 +1,20 @@
 apply plugin: 'distribution'
 
+configurations {
+    testCompile.extendsFrom compileOnly
```

Comment:
Don't you have the same problem if someone adds to compile a dependency this test module needs?

IMHO, it is better to stay with the default/convention to avoid surprises as much as possible.

If someone forgets to add a dependency, tests will fail. Right? Why isn't it good enough?